src/network/hvac_data.rs

```diff
@@ -0,0 +1,252 @@
+use packed_struct::prelude::{
+    packed_bits, Integer, PackedStruct, PackedStructSlice, PrimitiveEnum_u8,
+};
+
+use crate::{network::util::compute_generic_checksum, traits::CommandTrait};
+
+/// The type of command to send to the unit.
+#[derive(PrimitiveEnum_u8, Debug, Copy, Clone)]
+pub enum HvacDataCommand {
+    /// Set New State
+    SetState = 0x00,
+    /// Get Current State
+    GetState = 0x01,
+    /// Obtain air conditioner basic information
+    GetAcInfo = 0x02,
+}
+
+/// Enumerates modes.
+#[derive(PrimitiveEnum_u8, Clone, Copy, Debug)]
+pub enum HvacMode {
+    Auto = 0,
+    Cool = 1,
+    Dry = 2,
+    Heat = 3,
+    Fan = 4,
+}
+
+/// Enumerates fan speed.
+#[derive(PrimitiveEnum_u8, Clone, Copy, Debug)]
+pub enum HvacSpeed {
+    None = 0,
+    High = 1,
+    Mid = 2,
+    Low = 3,
+    Auto = 5,
+}
+
+/// Enumerates presets.
+#[derive(PrimitiveEnum_u8, Clone, Copy, Debug)]
+pub enum HvacPreset {
+    Normal = 0,
+    Turbo = 1,
+    Mute = 2,
+}
+
+/// Enumerates horizontal swing.
+#[derive(PrimitiveEnum_u8, Clone, Copy, Debug)]
+pub enum HvacSwHoriz {
+    LeftFix = 2,
+    LeftRightFix = 7,
+    RightFix = 6,
+    RightFlap = 5,
+    On = 0,
+    Off = 1,
+}
+
+/// Enumerates vertical swing.
+#[derive(PrimitiveEnum_u8, Clone, Copy, Debug)]
+pub enum HvacSwVert {
+    On = 0,
+    Pos1 = 1,
+    Pos2 = 2,
+    Pos3 = 3,
+    Pos4 = 4,
+    Pos5 = 5,
+    Off = 7,
+}
+
+/// A struct with air conditioner state.
+#[derive(PackedStruct, Debug)]
+#[packed_struct(bit_numbering = "msb0", size_bytes = "13")]
+pub struct AirCondState {
```

Comment:
Could you add some more docs for what these fields mean? Some are self explanatory, but I'm unsure what something like `ifeel` or `display` might mean.

Reply:
This also applies to the other structs in this file.

Reply:
Sure, updated the PR to cover this.